### PR TITLE
Add in Jira/Ticket reference to commit-msg

### PR DESCRIPTION
**Pull request by davidcollom**
Add in `commit-msg` style hooks along with a jira/ticket reference hook

### CODE REVIEW

**Comment by greg0ire** [comment]
Looks great!

Can you make it more generic though? Like, not tied to jira? And please add some documentation describing what this does.

**Comment by greg0ire** [comment]
Also, what happens/should happen when you have several commits on the branch?

**Comment by davidcollom** [comment]
Hey, I've squashed my commits and removed references of Jira into a `ticketref`plugin.

The expected behavior of this plugin is to prefix the commit message with the ticket reference from the branch name to ensure that the correlation of commit's against a specific feature/ticket/incident/issue etc.

File: docs/hooks/ticketref.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Description
+
+Prefixes the commit-msg with the ticket reference, this allows for some workflows to be able to correlate a number of commit's against a specific feature/ticket/incident/issue etc once merged.
```
Comment (greg0ire) [inline docs/hooks/ticketref.md]: - Please wrap after 80 chars (`gqq` in vim)
- "commit's" => "commits"

File: docs/hooks/ticketref.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+a branch of `ABC-1234-hello-world` would turn a commit of `git commit -m "Hello World"` into `[ABC-1234] Hello World`
```
Comment (greg0ire) [inline docs/hooks/ticketref.md]: Please capitalize the first word and put a dot at end of sentence.

File: docs/hooks/ticketref.md (hunk 1 above)
Comment (greg0ire) [inline docs/hooks/ticketref.md]: Also please add a dot after `etc`, it's an abbreviation.

File: docs/hooks/ticketref.md (hunk 2 above)
Comment (greg0ire) [inline docs/hooks/ticketref.md]: This has the big drawback of making the first line longer. The first line is supposed to be less than 50 chars. Would it be possible to produce something like this instead:

```
Hello world

Refs ABC-1234
```

**Comment by greg0ire** [comment]
cc @juliendufresne :P

**Comment by Edo78** [comment]
This looks interesting for my use case. Any chance it could be updated and accepted?

**Comment by greg0ire** [comment]
@Edo78 sure! You can make your own PR and address the issues, I will merge it!

**Comment by Edo78** [comment]
I feel a bit uneasy at copying the code of @davidcollom 
If he could give me his permission I'll be glad to do it

**Comment by davidcollom** [comment]
Hey, really sorry I completely forgot about the PR.

I may have some time today to update the PR based off the comments above, I don't see them taking much time.

**Comment by davidcollom** [comment]
@Edo78 / @greg0ire - I've updated this PR based off the comments you left prior

I added some additional configuration for this hook to support both requirements (Pre/Post reference locations).

**Pull request closed without merging**